clients/spark/core/src/main/scala/io/treeverse/clients/ApiClient.scala

```diff
@@ -27,8 +32,22 @@ private object ApiClient {
               uri.getQuery,
               uri.getFragment
              )
-    else
+    } else if (storageType == StorageTypeAzure) {
+
+      /** regex to catch url of type: https://StorageAccountName.blob.core.windows.net/Container[/BlobName] and convert it to abfs url */
+      val regex = "https?://([^.]+)(\\.[^/]+)(?:/([^/]+)(.+))?"
```

Comment:
These regexes ("regexen") would be easier to understand ("grok") if they did _not_ include the periods inside the matches.  Then you'd have to add the periods back in in the replacement string, but it would probably be easier for me to understand.